resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/StreamingCompatibilitySuite.scala

```diff
@@ -0,0 +1,214 @@
+        .endContainer()
+        .endSpec()
+        .done()
+      Eventually.eventually(TIMEOUT, INTERVAL) {
+        assert(getRunLog.contains("spark-streaming-kube"), "The application did not complete.")
+      }
+    } finally {
+      // Have to delete the service manually since it doesn't have an owner reference
+      kubernetesTestComponents
+        .kubernetesClient
+        .services()
+        .inNamespace(kubernetesTestComponents.namespace)
+        .delete(driverService)
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark streaming in cluster mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    try {
+      runSparkJVMCheckAndVerifyCompletion(
+        mainClass = "org.apache.spark.examples.streaming.NetworkWordCount",
+        appArgs = Array[String](host, port.toString),
+        expectedJVMValue = Seq("spark-streaming-kube"))
+    } finally {
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark structured streaming in cluster mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    try {
+      runSparkJVMCheckAndVerifyCompletion(
+        mainClass = "org.apache.spark.examples.sql.streaming.StructuredNetworkWordCount",
+        appArgs = Array[String](host, port.toString),
+        expectedJVMValue = Seq("spark-streaming-kube"))
+    } finally {
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark structured streaming in client mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    val driverService = driverServiceSetup
+    try {
+      setupSparkStreamingPod(driverService.getMetadata.getName)
+        .addToArgs("sql.streaming.StructuredNetworkWordCount")
+        .addToArgs(host, port.toString)
+        .endContainer()
+        .endSpec()
+        .done()
+
+      val TIMEOUT = PatienceConfiguration.Timeout(Span(3, Minutes))
+      Eventually.eventually(TIMEOUT, INTERVAL) {
+        assert(getRunLog.contains("spark-streaming-kube"),
+          "The application did not complete.")
+      }
+    }
+    finally {
+      // Have to delete the service manually since it doesn't have an owner reference
+      kubernetesTestComponents
+        .kubernetesClient
+        .services()
+        .inNamespace(kubernetesTestComponents.namespace)
+        .delete(driverService)
+      serverSocket.close()
+    }
+  }
+
+  private def getRunLog: String = kubernetesTestComponents.kubernetesClient
+    .pods()
+    .withName(driverPodName)
+    .getLog
+
+  private def setupSparkStreamingPod(driverServiceName: String) = {
+    val labels = Map("spark-app-selector" -> driverPodName)
+    testBackend
+      .getKubernetesClient
+      .pods()
+      .inNamespace(kubernetesTestComponents.namespace)
+      .createNew()
+      .withNewMetadata()
+      .withName(driverPodName)
+      .withLabels(labels.asJava)
+      .endMetadata()
+      .withNewSpec()
+      .withServiceAccountName(kubernetesTestComponents.serviceAccountName)
```

Comment:
The flakiness of the tests seems to be in relation to this service account. Investigate the necessary rbac.yml that would need to be set to ensure that these failures don't come up?

Reply:
I don't believe this was solved in the Client mode tests, despite the addition of the `spark-rbac.yml` I think this might require investigation outside of the scope of this PR.

Reply:
I have run these tests on my own setup of minikube, and I am unable to reproduce the failure that occurred on jenkins. It is possible that is related to how minikube is setup on jenkins.